honeybadgermpc/sequencer.py

```diff
@@ -0,0 +1,25 @@
+import heapq
+
+
+class Sequencer(object):
+    def __init__(self):
+        self.heap = []
+        self.values = set()  # Use this to ensure a duplicate is not added
+        self.next = 0
+
+    def get(self):
+        assert self.is_next_available()
+        value = heapq.heappop(self.heap)
+        self.values.remove(value[0])
+        self.next += 1
+        return value
+
+    def is_next_available(self):
+        return len(self.heap) > 0 and self.heap[0][0] == self.next
+
+    def add(self, value):
```

Comment:
It seems like the interface `add(self, sequence_number, value)` would be a good fit here?

Reply:
I had thought about it but if you take a look at the `heapq.heappush` method it follows the same design. I would have to create a tuple to club the values anyway to push them on to the `heap`. Having it as a tuple, also means that I can push the AVSSed value directly without having to extract the sequence number from it first and then insert it.